lib/rules/prefer-separate-static-class.js

```diff
@@ -0,0 +1,245 @@
+/**
+ * @author Flo Edelmann
+ * See LICENSE file in root directory for full license.
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const { defineTemplateBodyVisitor, getStringLiteralValue } = require('../utils')
+
+// ------------------------------------------------------------------------------
+// Helpers
+// ------------------------------------------------------------------------------
+
+/**
+ * @param {ASTNode} node
+ * @returns {node is Literal | TemplateLiteral}
+ */
+function isStringLiteral(node) {
+  return (
+    (node.type === 'Literal' && typeof node.value === 'string') ||
+    (node.type === 'TemplateLiteral' && node.expressions.length === 0)
+  )
+}
+
+/**
+ * @param {VReference[]} references
+ * @param {Identifier} identifier
+ * @returns {boolean}
+ */
+function referencesInclude(references, identifier) {
+  return references.some((reference) => reference.id === identifier)
+}
+
+/**
+ * @param {Expression | VForExpression | VOnExpression | VSlotScopeExpression | VFilterSequenceExpression} expressionNode
+ * @param {VReference[]} references
+ * @returns {(Literal | TemplateLiteral | Identifier)[]}
+ */
+function findStaticClasses(expressionNode, references) {
+  if (isStringLiteral(expressionNode)) {
+    return [expressionNode]
+  }
+
+  if (expressionNode.type === 'ArrayExpression') {
+    return expressionNode.elements.flatMap((element) => {
+      if (element === null || element.type === 'SpreadElement') {
+        return []
+      }
+      return findStaticClasses(element, references)
+    })
+  }
+
+  if (expressionNode.type === 'ObjectExpression') {
+    return expressionNode.properties.flatMap((property) => {
+      if (
+        property.type === 'Property' &&
+        property.value.type === 'Literal' &&
+        property.value.value === true &&
+        (isStringLiteral(property.key) ||
+          (property.key.type === 'Identifier' &&
+            !referencesInclude(references, property.key)))
```

Comment:
You probably just need to check `computed`.

https://github.com/estree/estree/blob/master/es2015.md#:~:text=boolean%3B%0A%20%20%20%20shorthand%3A%20boolean%3B-,computed,-%3A%20boolean%3B%0A%7D

```suggestion
            !property.computed
```

Reply:
Cool, that simplifies that bit quite a bit!

Reply:
Fixed in [`bd51d1e`](https://github.com/vuejs/eslint-plugin-vue/pull/1729/commits/bd51d1e88b00eef1b71bdd8afe1fc14c79ff7f4b).